### PR TITLE
Toggle unity logs with debug setting; Distant fog fix; Small fixes.

### PR DESCRIPTION
**Pull request by artumino**
General optimization by removing Reflection from elements that run on the update loop, reduced log spam from Nomai Projection Pools and fixed and exception introduced by the scope camera update

### CODE REVIEW

**Comment by Raicuparta** [comment]
We could probably remove the need for this caching and reduce the complexity of anything that relies on reflection by using this:

https://github.com/CabbageCrow/AssemblyPublicizer

This would make all fields public, and it should only need to be done during development. Drawback is that it adds an extra step to the development setup.

**Comment by Raicuparta** [comment]
Here is what it looks like with the publicizer: https://github.com/Raicuparta/nomai-vr/pull/394
I only replaced what I could find with a few regular expressions, there's probably more usages of reflection that could be removed by this. What do you think?

File: NomaiVR/Player/PlayerBodyPosition.cs [hunk 1]
```diff
@@ -164,6 +165,15 @@ private static void PatchTurning(Action<Quaternion> rotationSetter)
...
                     }
                 }
+
+                private static bool PatchOWCameraFOV(OWCamera __instance)
+                {
+                    //Prevents changing the fov of VR cameras
+                    //This prevents log spams in projection pools
+                    if (__instance.mainCamera.stereoEnabled)
+                        return false;
+                    return true;
```
Comment (Raicuparta) [inline NomaiVR/Player/PlayerBodyPosition.cs:175]: simpler?
```suggestion
                    return !__instance.mainCamera.stereoEnabled;
```

File: NomaiVR/Tools/HoldSignalscope.cs [hunk 2]
```diff
@@ -156,7 +156,9 @@ private void UpdateSignalscipeZoom()
                 if (OWInput.IsNewlyPressed(InputLibrary.scopeView, InputMode.All) && ToolHelper.Swapper.IsInToolMode(ToolMode.SignalScope, ToolGroup.Suit))
                 {
                     _lens.gameObject.SetActive(!_lens.gameObject.activeSelf);
-                    _owLensCamera.SetEnabled(_lens.gameObject.activeSelf);
+
+                    if(_owLensCamera.gameObject != null)
```
Comment (Raicuparta) [inline NomaiVR/Tools/HoldSignalscope.cs:160]: So it's possible for a component to still exist while its gameObject is null?

Comment (artumino) [reply NomaiVR/Tools/HoldSignalscope.cs:160]: Yes, I don't know why but suiting up is a way for it to happen

Comment (Raicuparta) [reply NomaiVR/Tools/HoldSignalscope.cs:160]: Strange, I think it's only supposed to be null after it's destroyed. But if it works, sure.

**Review by Raicuparta — APPROVED** [review]

**Pull request merged**